Sources/Shared/Protocols/PathAccessible.swift

```diff
@@ -36,19 +37,50 @@ public extension PathAccessible {
       }
     }
 
-    return result as? JSONDictionary
+    return result as? T
   }
 
-  func path(keyPath: String) -> JSONDictionary? {
-    let kinds: [SubscriptKind] = keyPath.componentsSeparatedByString(".").map {
+  private func resolve<T>(path: String) -> T? {
+    let kinds: [SubscriptKind] = path.componentsSeparatedByString(".").map {
       if let index = Int($0) {
         return .Index(index)
       } else {
         return .Key($0)
       }
     }
 
-    return path(kinds)
+    return resolve(kinds)
+  }
+
+  private func extractKey(path: String) -> (key: String, keyPath: String) {
+    return (key: path.split(".").last!,
+            keyPath: Array(path.split(".").dropLast()).joinWithSeparator("."))
+  }
+
+  func path(path: [SubscriptKind]) -> JSONDictionary? {
+    return resolve(path)
+  }
+
+  func path(keyPath: String) -> JSONDictionary? {
+    return resolve(keyPath)
+  }
+
+  func path(keyPath: String) -> String? {
```

Comment:
We could do that refactoring in a different pull request and then bump to a new major version number, right? 😁